File: FlutterApp/lib/models/table_models/income.dart
class Income {
  int? id;
  String name;
  double incomeValue;
  DateTime initialDate;
  DateTime endDate;
  int timesRecurrence;
  int? recurrenceId;
  int isEndless;

  Income({
    this.id,
    required this.name,
    required this.incomeValue,
    required this.initialDate,
    required this.endDate,
    required this.recurrenceId,
    required this.timesRecurrence,
    required this.isEndless,
  });

  Map<String, dynamic> toMap() {
    final incomeMap = Map<String,dynamic>();
    if(id != null) {
      incomeMap['id'] = id;
    }
    incomeMap['name'] = name;
    incomeMap['incomeValue'] = incomeValue;
    incomeMap['initialDate'] = initialDate.millisecondsSinceEpoch;
    incomeMap['endDate'] = endDate.millisecondsSinceEpoch;
    incomeMap['recurrenceId'] = recurrenceId;
    incomeMap['timesRecurrence'] = timesRecurrence;
    incomeMap['isEndless'] = isEndless;

    return incomeMap;
  }
  factory Income.fromMap(Map<String, dynamic> row){
    return Income(
      id: row['id'],
      name: row['name'],
      incomeValue: row['incomeValue'],
      initialDate: DateTime.fromMillisecondsSinceEpoch(row['initialDate']),
      endDate: DateTime.fromMillisecondsSinceEpoch(row['endDate']),
      recurrenceId: row['recurrenceId'],
      timesRecurrence: row['timesRecurrence'],
      isEndless: row['isEndless']
      );
  }
}

File: FinanceApp.Shared/Entities/UserTables/Income.cs
using FinanceApp.Shared.Enum;
using System.ComponentModel.DataAnnotations;

namespace FinanceApp.Shared.Entities.UserTables
{
    public class Income : UserTable
    {
        [Required]
        public string Name { get; set; } = String.Empty;
        [Required]
        public ERecurrence Recurrence { get; set; }
        public double Amount { get; set; }
        public DateTime InitialDate { get; set; }
        public DateTime? EndDate { get; set; }
        public bool IsEndless { get; set; }
        public int? TimesRecurrence { get; set; }

        public override void CheckInput()
        {
            if (Recurrence != ERecurrence.Once && EndDate == null && !IsEndless && (TimesRecurrence == null || TimesRecurrence == 0))
                throw new Exception("A quantidade de repetições deve ser maior que zero para o tipo de recorrência selecionado");

            else if (Amount <= 0.00)
                throw new Exception("O valor deve ser maior do que zero");
        }
    }
}